File: SpeedRunAppImport.Interfaces/Services/ISettingService.cs
using System;
using System.Collections.Generic;
using SpeedRunAppImport.Model.Entity;
using System.Threading.Tasks;

namespace SpeedRunAppImport.Interfaces.Services
{
    public interface ISettingService
    {
        SettingEntity GetSetting(string name);
        void UpdateSetting(string name, string value);
        void UpdateSetting(string name, DateTime value);
        void UpdateSetting(string name, int value);
        void UpdateSetting(SettingEntity setting);
        string GetTwitchToken();
        bool GenerateAndMoveSitemapXml();
    }
}

File: src/main/java/speedrunappimport/interfaces/services/ISettingService.java
package speedrunappimport.interfaces.services;

import java.time.Instant;

import speedrunappimport.model.entity.Setting;

public interface ISettingService
{
	Setting GetSetting(String name);
	void UpdateSetting(String name, Instant value);
	void UpdateSetting(String name, String value);
	void UpdateSetting(String name, Integer value);
	void UpdateSetting(Setting setting);
}
